src/models/chen2021Fitness/Chen2021FitnessWidget.tsx

```diff
@@ -0,0 +1,381 @@
+import { Widget } from '../../widgets/Widget';
+import { ZodQueryEncoder } from '../../helpers/query-encoder';
+import { SampleSelectorSchema } from '../../helpers/sample-selector';
+import * as zod from 'zod';
+import React, { useEffect, useState } from 'react';
+import { get } from '../../services/api';
+import {
+  Chen2021FitnessRequestSchema,
+  Chen2021FitnessResponseSchema,
+  ValueWithCISchema,
+} from './chen2021Fitness-types';
+import styled from 'styled-components';
+import { Col, Form, Button } from 'react-bootstrap';
+import { Plot } from '../../components/Plot';
+
+const ContainerPropsSchema = SampleSelectorSchema;
+type ContainerProps = zod.infer<typeof SampleSelectorSchema>;
+
+const ResultsPropsSchema = zod.object({
+  request: Chen2021FitnessRequestSchema,
+});
+type ResultsProps = zod.infer<typeof ResultsPropsSchema>;
+
+type Chen2021FitnessRequest = zod.infer<typeof Chen2021FitnessRequestSchema>;
+type Chen2021FitnessResponse = zod.infer<typeof Chen2021FitnessResponseSchema>;
+
+const getData = async (
```

Comment:
Would you prefer moving it to `api.ts` or should we leave it here?

Reply:
I dont know

Reply:
Let's keep it separate from other API functions for now.